src/intervals/arithmetic/basic.jl

```diff
@@ -2,120 +2,95 @@
 # Section 9.1 of the IEEE Standard 1788-2015 and required for set-based flavor
 # in Section 10.5.3
 
-+(a::Interval) = a # not in the IEEE Standard 1788-2015
++(a::BareInterval) = a # not in the IEEE Standard 1788-2015
+
++(a::Interval) = a
 
 """
+    +(a::BareInterval, b::BareInterval)
     +(a::Interval, b::Interval)
-    +(a::Interval, b::Real)
-    +(a::Real, b::Interval)
 
 Implement the `add` function of the IEEE Standard 1788-2015 (Table 9.1).
 """
-function +(a::Interval{T}, b::Interval{T}) where {T<:NumTypes}
+function +(a::BareInterval{T}, b::BareInterval{T}) where {T<:NumTypes}
     isempty_interval(a) && return a
     isempty_interval(b) && return b
     return @round(T, inf(a) + inf(b), sup(a) + sup(b))
 end
++(a::BareInterval, b::BareInterval) = +(promote(a, b)...)
 
-+(a::Interval, b::Interval) = +(promote(a, b)...)
-
-function +(a::Interval{T}, b::T) where {T<:NumTypes}
-    isempty_interval(a) && return a
-    isfinite(b) || return emptyinterval(T)
-    return @round(T, inf(a) + b, sup(a) + b)
+function +(a::Interval, b::Interval)
+    r = bareinterval(a) + bareinterval(b)
+    d = min(decoration(a), decoration(b), decoration(r))
+    t = guarantee(a) & guarantee(b)
+    return _unsafe_interval(r, d, t)
 end
 
-+(a::Interval{T}, b::Real) where {T<:NumTypes} = a + interval(T, b)
-
-+(a::Real, b::Interval{T}) where {T<:NumTypes} = b + a
-
 """
+    -(a::BareInterval)
     -(a::Interval)
 
 Implement the `neg` function of the IEEE Standard 1788-2015 (Table 9.1).
 """
--(a::Interval{T}) where {T<:NumTypes} = unsafe_interval(T, -sup(a), -inf(a))
+-(a::BareInterval{T}) where {T<:NumTypes} = _unsafe_bareinterval(T, -sup(a), -inf(a))
+
+-(a::Interval) = _unsafe_interval(-bareinterval(a), decoration(a), guarantee(a))
 
 """
+    -(a::BareInterval, b::BareInterval)
     -(a::Interval, b::Interval)
-    -(a::Interval, b::Real)
-    -(a::Real, b::Interval)
 
 Implement the `sub` function of the IEEE Standard 1788-2015 (Table 9.1).
 """
-function -(a::Interval{T}, b::Interval{T}) where {T<:NumTypes}
+function -(a::BareInterval{T}, b::BareInterval{T}) where {T<:NumTypes}
     isempty_interval(a) && return a
     isempty_interval(b) && return b
     return @round(T, inf(a) - sup(b), sup(a) - inf(b))
 end
+-(a::BareInterval, b::BareInterval) = -(promote(a, b)...)
 
--(a::Interval, b::Interval) = -(promote(a, b)...)
-
-function -(a::Interval{T}, b::T) where {T<:NumTypes}
-    isempty_interval(a) && return a
-    return @round(T, inf(a) - b, sup(a) - b)
+function -(a::Interval, b::Interval)
+    r = bareinterval(a) - bareinterval(b)
+    d = min(decoration(a), decoration(b), decoration(r))
```

Comment:
Is there a decoration associated to `r`? Or is it `com` by default? I think we can spare `decoration(r)` in this line.

Reply:
Thanks for the explanation! Forget my comment...